src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -1812,6 +1812,17 @@ protected boolean executeProgram(int dataSetNumber) {
             }
 
             log.log(Log.DEBUG, "before substitution: " + cmdline);
+
+            /**
+             * Special substitution for entry_point.
+             * We only do this substition for execution of the run, not compiles or validators, so it can't
+             * be done by substitutionAllStrings without refactoring.
+             */
+            if (run.getEntryPoint() != null) {
```

Comment:
The 2022-07 CLICS spec defines the type of the "entry point" property as `string ?` (meaning a string which might be null).  The spec further gives explicit examples of what must be provided as the entry point for Python (the name of the file which must be run), and for Java and Kotlin (the fully-qualified class name, including any package prefixes).  The spec further explicitly states that "For C and C++ it must be `null`".  However, the spec is silent about what 
must/may/should/should not be provided for the entry point for any _other_ language.  I believe that it's possible that an implementer could legitimately, for example, pass in the _empty string_ as an indication that "no entry point is specified" for some (other) language.  

If that were to happen, I believe the above test would cause PC2 to fail -- because it would NOT see a "null" value returned from `run.getEntryPoint()` (because "the empty string" is not the same as "null") -- so it would (improperly) replace the main program command name with an empty string.   An entry point which was not null and also not empty but contained only whitespace would cause a similar problem.

For defensive programming (i.e., to avoid troubles down the road) I think the above test should be something like 
```java
String ep = run.getEntryPoint() ;
if ( ep != null  &&  !ep.isEmpty()  &&  !ep.isBlank() ) {
...  //do the substitution
```

Reply:
Note: there is no isBlank method in Java 8

Reply:
I recommend using StringUtilities.isBlank(s), that way when isBlank is available that change can be reflected